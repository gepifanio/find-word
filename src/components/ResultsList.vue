<template>
  <div class="bg-white shadow-xl rounded-lg w-1/2">
    <ul class="divide-y divide-gray-300" v-for="item in data" :key="item">
      <li class="p-4 hover:bg-gray-50">{{ item }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    data: {
      type: Array,
      default: () => ['']
    }
  }
};
</script>
