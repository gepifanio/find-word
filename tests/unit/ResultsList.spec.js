import { shallowMount } from '@vue/test-utils';
import ResultsList from '@/components/ResultsList.vue';

describe('ResultsList.vue', () => {
  const data = ['dog', 'god'];

  it('should match snapshot', () => {
    const result = shallowMount(ResultsList, {
      propsData: {
        data,
      }
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('renders props.data when passed', () => {
    const wrapper = shallowMount(ResultsList, {
      propsData: {
        data
      }
    });

    expect(typeof wrapper.props().data).toBe('object');
    expect(wrapper.props().data).toBe(data);
    expect(wrapper.find('[data-test="result-data-item"]').text()).toEqual(
      'dog'
    );
  });
});
