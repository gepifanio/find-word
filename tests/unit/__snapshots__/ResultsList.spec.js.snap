// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList.vue should match snapshot 1`] = `
<div
  class="bg-white shadow-xl rounded-lg w-1/2"
>
  <ul
    class="divide-y divide-gray-300"
  >
    <li
      class="p-4 hover:bg-gray-50"
      data-test="result-data-item"
    >
      dog
    </li>
  </ul>
  <ul
    class="divide-y divide-gray-300"
  >
    <li
      class="p-4 hover:bg-gray-50"
      data-test="result-data-item"
    >
      god
    </li>
  </ul>
</div>
`;
